packages/cw3/src/helpers.rs

```diff
@@ -0,0 +1,80 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{to_binary, Api, CanonicalAddr, CosmosMsg, HumanAddr, StdResult, WasmMsg};
+
+use crate::msg::{Cw3HandleMsg, Vote};
+use cw0::Expiration;
+
+/// Cw3Contract is a wrapper around HumanAddr that provides a lot of helpers
+/// for working with this.
+///
+/// If you wish to persist this, convert to Cw3CanonicalContract via .canonical()
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Cw3Contract(pub HumanAddr);
+
+impl Cw3Contract {
+    pub fn addr(&self) -> HumanAddr {
+        self.0.clone()
+    }
+
+    /// Convert this address to a form fit for storage
+    pub fn canonical<A: Api>(&self, api: &A) -> StdResult<Cw3CanonicalContract> {
+        let canon = api.canonical_address(&self.0)?;
+        Ok(Cw3CanonicalContract(canon))
+    }
+
+    pub fn encode_msg(&self, msg: Cw3HandleMsg) -> StdResult<CosmosMsg> {
+        Ok(WasmMsg::Execute {
+            contract_addr: self.addr(),
+            msg: to_binary(&msg)?,
+            send: vec![],
+        }
+        .into())
+    }
+
+    /// helper doesn't support custom messages now
+    pub fn proposal<T: Into<String>, U: Into<String>>(
+        &self,
+        title: T,
+        description: U,
+        msgs: Vec<CosmosMsg>,
```

Comment:
This does not support custom message types

Reply:
There is a comment to that effect on line 36 ( ` /// helper doesn't support custom messages now` )

Reply:
I added a FIXME for this